File: Selenium/Activities/Activity9.py
from selenium.webdriver import ActionChains, Keys
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set up the actions object
    actions = ActionChains(driver)
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/ajax")
    print("Title of the page is: ", driver.title)
    # Find Change content button and click
    driver.find_element(By.CLASS_NAME, "violet").click()
    wait = WebDriverWait(driver, 10)
    text = wait.until(EC.presence_of_element_located((By.TAG_NAME, "h1"))).text
    print("Message that appears on the page: ", text)
    wait.until(EC.text_to_be_present_in_element((By.TAG_NAME, "h3"), "I'm late!"))
    text1 = driver.find_element(By.TAG_NAME, "h3").text
    print("The new message on the page: ", text1)

File: Selenium/Activities/Activity9.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/ajax");
        System.out.println("Title of the page: " + driver.getTitle());
        // Find and click the "Change content" button on the page
        driver.findElement(By.className("violet")).click();
        // Wait for the text to say "HELLO!"
        WebDriverWait wait = new WebDriverWait (driver, Duration.ofSeconds(20));
        String text = wait.until(ExpectedConditions.presenceOfElementLocated(By.tagName("h1"))).getText();
        System.out.println("Message that appears on the page: " +text);
        // Wait for the text to change to contain "I'm late!"
        String h3text = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath("//h3[contains(text(),'late')]"))).getText();
        System.out.println("The new message on the page: " +h3text);
        driver.quit();
    }
}
